ipfsapi/encoding.py

```diff
@@ -18,21 +22,62 @@
 class Encoding(object):
     """Abstract base for a data parser/encoder interface.
     """
+    __metaclass__ = abc.ABCMeta
 
-    def parse(self, string):
-        """Parses string into corresponding encoding.
+    @abc.abstractmethod
+    def parse_partial(self, raw):
```

Comment:
I see now. You have to wait to throw the parsing exceptions. I'm basically leaving this comment for my own reference at this point.